rotkehlchen/db/custom_assets.py

```diff
@@ -0,0 +1,105 @@
+        return [CustomAsset.deserialize_from_db(entry) for entry in cursor]
+
+    def get_custom_assets_and_limit_info(
+            self,
+            cursor: 'DBCursor',
+            filter_query: CustomAssetsFilterQuery,
+    ) -> Tuple[List[CustomAsset], int]:
+        """Retrieves all custom assets with pagination."""
+        custom_assets = self._get_custom_assets(
+            cursor=cursor,
+            filter_query=filter_query,
+        )
+        query, bindings = filter_query.prepare(with_pagination=False)
+        query = 'SELECT COUNT(*) from custom_assets ' + query
+        total_found_result = cursor.execute(query, bindings).fetchone()[0]
+        return custom_assets, total_found_result
+
+    def add_custom_asset(self, custom_asset: CustomAsset) -> str:
+        """Adds a custom new custom asset into `assets` and `custom_assets` table.
+        """
+        with self.global_db.conn.write_ctx() as global_db_write_cursor:
+            global_db_write_cursor.execute(
+                'INSERT INTO assets(identifier, type) VALUES (?, ?)',
+                (custom_asset.identifier, AssetType.CUSTOM_ASSET.serialize_for_db()),
+            )
+            global_db_write_cursor.execute(
+                'INSERT INTO custom_assets(identifier, name, notes, type) VALUES(?, ?, ?, ?)',  # noqa: E501
+                custom_asset.serialize_for_db(),
+            )
+            with self.db.user_write() as db_write_cursor:
+                self.db.add_asset_identifiers(db_write_cursor, [custom_asset.identifier])
+        return custom_asset.identifier
+
+    def edit_custom_asset(self, custom_asset: CustomAsset) -> None:
+        """Edits a custom asset.
+        May raise:
+        - InputError if editing a custom asset that does not exist or name is already used.
+        """
+        result = self.check_custom_asset_exists(
+            name=custom_asset.name,
+            identifier=custom_asset.identifier,
+        )
+        if result is True:
+            raise InputError(
+                f'Tried to edit custom asset with id {custom_asset.identifier} and name '
+                f'{custom_asset.name} but name is already in use.',
+            )
+        with self.global_db.conn.write_ctx() as write_cursor:
+            write_cursor.execute(
+                'UPDATE custom_assets SET name=?, notes=?, type=? WHERE identifier=?',
+                (
+                    custom_asset.name,
+                    custom_asset.notes,
+                    custom_asset.custom_asset_type,
+                    custom_asset.identifier,
+                ),
+            )
+            if write_cursor.rowcount == 0:
+                raise InputError(
+                    f'Tried to edit custom asset with id {custom_asset.identifier} and name '
+                    f'{custom_asset.name} but it was not found',
+                )
+
+    def check_custom_asset_exists(self, name: str, identifier: Optional[str] = None) -> bool:
```

Comment:
> I don't think that name has to be unique. Why do you think so?

I answered this here but a tldr, to avoid confusion.

> Even if we will agree that name has to be unique I would better add it as a db constraint and catch errors instead of manual checking

Yeah, if I add a unique constraint to the schema, I intend to remove this function.